classes/property.py

```diff
@@ -46,8 +46,7 @@ def property_value(property, data):
 
 def random_value(values):
     if values:
-        v = random()
-        for (value, cutoff) in values:
-            if v < cutoff:
-                return value
+        cutoff = random()
+        return next((value for value, limit in values if cutoff >= limit),
```

Comment:
This seems like it's not selecting a random value, it's getting the next value whose limit does not exceed some cutoff, which is randomly selected. So perhaps the name it wants is `next_value_below_random_threshold` or something, which suggests that the underlying behavior is `next_value_below_cutoff` and it takes `cutoff` as a parameter. 
Also, I think the `if values` check is not needed:

```
>>> next((value for value in []), None)
>>> next((value for value in []), None) is None
True
>>> 
```